### PR TITLE
WIP Refactored the Design  / Renamed files (see commits) 

### PR DESCRIPTION
**Pull request by oarzani**
Working on the filterfunction:
Need to set a connection between the PriceBar.js (see components) that shall define the Price in the Modal **AND** the getStudioByFilters -Function. 

Outsourced the filter-function to "getStudioByFilters.js"

### CODE REVIEW

File: src/getStudioByFilters.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { studios } from "./api/studio";
+
+export function getStudioByFilters(filters) {
+  // const selectedfilters = Object.keys(filters); // filters === ['flexx', 'six', 'twelve']
+
+  return studios.filter(studio => {
+    console.log(filters, studio);
+    if (filters.price) {
+      const minPrice = Math.min(...studio.price);
```
Comment (oarzani) [inline src/getStudioByFilters.js]: In this part the array should be filtered by the price

File: src/components/Modal.js [hunk 2]
```diff
@@ -93,13 +97,8 @@ export default function Modal({
...
         </BadgeContainer>
-        <Title>Choose Price</Title>
-        <RangeBar
-          filters={filters}
-          active={filters[badgeKeys[10]]}
-          setFilters={setFilters}
-          onChange={handleChange}
-        />
+        <Title>Choose Price-Limit</Title>
+        <PriceBar />
```
Comment (oarzani) [inline src/components/Modal.js]: Tried to move the filters setFilter React State down to the PriceBar. The Price-Bar had its own  default state which was replaced by [filters, setFilters]. After that the Price Bar was not working in the Modal.

File: src/components/PriceBar.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const Wrapper = styled.div`
   padding: 20px;
 `;
 
-export default function RangeBar() {
+export default function PriceBar() {
   const [selectedValue, setSelectedValue] = useState("");
```
Comment (oarzani) [inline src/components/PriceBar.js]: This state was replaced by the states filters and setFilters i moved down. Need here the connection between price bar and the state of the filters.

File: src/pages/Gyms.js [hunk 4]
```diff
@@ -37,15 +38,14 @@ const defaultFilters = {
   six: false,
   twelve: false,
   twentyfour: false,
-  price: 10
+  price: 90
```
Comment (oarzani) [inline src/pages/Gyms.js]: added price to the state which was moved down to the PriceBar

File: src/components/Modal.js [hunk 5]
```diff
@@ -93,13 +97,8 @@ export default function Modal({
             onBadgeClick={onBadgeClick}
           />
         </BadgeContainer>
-        <Title>Choose Price</Title>
-        <RangeBar
-          filters={filters}
-          active={filters[badgeKeys[10]]}
```
Comment (oarzani) [inline src/components/Modal.js]: Tried it out with that before i renamed the bar.

File: src/components/Modal.js [hunk 6]
```diff
@@ -62,14 +63,17 @@ export default function Modal({
...
     <>
       <Container show={show}>
-        <Title>Choose Contract Duration</Title>
+        <Title>
+          Choose Contract Duration <br />
+          (Months)
+        </Title>
         {/* <DurationBar /> */}
         <BadgeContainer>
           <BadgeDuration
```
Comment (marwinburesch) [inline src/components/Modal.js]: you could use `.map()` on `badgeKeys` to increase efficiency here, same for the next `<BadgeContainer>`

File: src/components/Modal.js (hunk 2 above)
Comment (marwinburesch) [reply src/components/Modal.js]: one of the next steps, i.e. a branch, would be to add props and finish this component

File: src/pages/Details.js [hunk 7]
```diff
@@ -74,14 +80,17 @@ export default function Details({ match }) {
         </ImageWrapper>
         <Wrapper>
```
Comment (marwinburesch) [inline src/pages/Details.js]: please continue your convention to specify what the wrapper is for in the name, like you did with `<ImageWrapper>`

File: src/components/PriceBar.js (hunk 3 above)
Comment (marwinburesch) [reply src/components/PriceBar.js]: With "the states" you probably mean the state `filters` and its corresponding function to change the states value `setFilters`

File: src/components/card/CardComponent.js [hunk 8]
```diff
@@ -1,24 +1,23 @@
...
-  max-height: 200px;
 `;
 
 //To-Do: Link property to Details
-export default function CardComponent({ studio }) {
+export default function CardComponent({ studio, props, name }) {
   return (
     <CardWrapper>
+      <StyledLink studio={studio} to={`/gyms/${name}`} />
       <CardElementPic
```
Comment (marwinburesch) [inline src/components/card/CardComponent.js]: I think this component is receiving at least one unused property

File: src/pages/Gyms.js (hunk 4 above)
Comment (marwinburesch) [reply src/pages/Gyms.js]: Regarding: "was moved down...", you probably mean "is passed on to... component"

File: src/components/Header.js [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
 import { Link } from "react-router-dom";
 
 const HeaderDiv = styled.header`
-  height: 65px;
+  height: 60px;
   background: rgb(78, 203, 224);
   width: 100%;
   display: flex;
   justify-content: center;
 `;
```
[the remark this answers is not in the document]
Comment (oarzani) [reply src/components/Header.js]: The wrapper is necessary to have the margin of the sing outbounded elements

File: src/components/Modal.js [hunk 10]
```diff
@@ -25,6 +25,7 @@ const SubmitButton = styled.div`
   position: absolute;
   left: calc(50% - 22px);
```
[the remark this answers is not in the document]
Comment (oarzani) [reply src/components/Modal.js]: thx

File: src/components/accordion/AccordionContracts.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+const Content = styled.div`
+  display: ${props => (props.show ? "block" : "none")};
+  width: 100%;
+  color: #0001ff;
+`;
+
+const Arrow = styled.div`
+  align-self: right;
+  transform: ${props => (props.show ? "rotate(180deg)" : "rotate(0deg)")};
```
[the remark this answers is not in the document]
Comment (oarzani) [reply src/components/accordion/AccordionContracts.js]: Sorry but can you describe it more in detail please?

**Pull request merged**